src/routes/explore/+page.svelte

```diff
@@ -1 +1,286 @@
-Explore page
+<script lang="ts">
+    import { OpengraphGenerator } from "$functions/opengraph";
+    import { page } from "$app/stores";
+    import Search from "$icons/search.svelte";
+    import Chevron from "$icons/chevron.svelte";
+    import Preference from "$icons/preference.svelte";
+    import Circle from "$icons/circle.svelte";
+    import { trending_animes } from "$data/mock/trending";
+    import ImageLoader from "$components/shared/image/image_loader.svelte";
+    import { blur } from "svelte/transition";
+
+    /* Anime cards scroll */
+    // no:of items to show on each scroll
+    let SHOW_NEW_CARDS_COUNT = 2,
+        trending_animes_scroll_element: HTMLElement,
+        popular_animes_scroll_element: HTMLElement,
+        last_scrolled: "trending" | "popular";
+
+    const show_scroll_buttons: {
+        [key in typeof last_scrolled]: {
+            left: boolean;
+            right: boolean;
+        };
+    } = {
+        trending: {
+            left: false,
+            right: true
+        },
+        popular: {
+            left: false,
+            right: true
+        }
+    };
```

Comment:
Honestly if you ask me i dont think we need this thing at all. We can do something with CSS to avoid this type of logic

Reply:
Okay... whatya think about this? this single function will handle show hide logic via css.
```ts
function handle_scroll(event: UIEvent) {
    const element = event.target as HTMLElement;
    const { scrollLeft, scrollWidth, clientWidth } = element;

    const left_scroll_button = element.offsetParent?.children[1].firstChild as HTMLElement;
    const right_scroll_button = element.offsetParent?.children[1].lastChild as HTMLElement;

    console.log(left_scroll_button);

    switch(scrollLeft + clientWidth) {
        // scroll is on utter right
        case scrollWidth:
            right_scroll_button.style.opacity = "0";
            break;
        // scroll is on utter left
        case clientWidth:
            left_scroll_button.style.opacity = "0";
            break;
        // scroll is somewhere middle
        default:
            left_scroll_button.style.opacity = "1";
            right_scroll_button.style.opacity = "1";
            break;
    }
}
```

Reply:
+- what we are looking for